lib/resolve-config.js

```diff
@@ -44,39 +43,41 @@ module.exports = (
     assets: assets ? castArray(assets) : assets,
     milestones: milestones ? castArray(milestones) : milestones,
     successComment,
-    proxy: getProxyConfiguration(defaultedGitlabUrl, HTTP_PROXY, HTTPS_PROXY),
+    proxy: getProxyConfiguration(defaultedGitlabUrl),
   };
 };
 
-function getProxyConfiguration(gitlabUrl, HTTP_PROXY, HTTPS_PROXY) {
+function getProxyConfiguration(gitlabUrl) {
   const sharedParameters = {
     keepAlive: true,
     keepAliveMsecs: 1000,
     maxSockets: 256,
     maxFreeSockets: 256,
     scheduling: 'lifo',
   };
+  const proxy = getProxyForUrl(gitlabUrl);
```

Comment:
I don't really like that we are now using the environment variables implicitly. This is unexpected and reduced testability. Can you change that back, maybe just copying the relevant parts for `getProxyForUrl`?

Reply:
I derived the relevant part then from the proxy-from-env package and changed it back so that no longer implicit environment variables are used.